test/r_tree/r_tree_test.dart

```diff
@@ -197,30 +213,105 @@ main() {
             tree.insert(item);
           }
         }
+        assertTreeHeightValidity(tree);
 
         var items = tree.search(Rectangle(0, 0, 50, 50));
         expect(items.length, equals(2500));
 
         data.forEach((RTreeDatum item) {
           tree.remove(item);
         });
+        assertTreeHeightValidity(tree);
 
         items = tree.search(Rectangle(0, 0, 50, 50));
         expect(items.length, equals(0));
 
         //test inserting after removal to ensure new root leaf node functions correctly
         tree.insert(RTreeDatum<String>(Rectangle(0, 0, 1, 1), 'New Initial Item'));
+        assertTreeHeightValidity(tree);
 
         items = tree.search(Rectangle(0, 0, 50, 50));
 
         items.forEach((datum) {
           expect(datum.value, equals('New Initial Item'));
         });
       });
+
+      test('remove all items and then reload', () {
+        final tree = RTree(3);
+
+        final items = <RTreeDatum<String>>[];
+        for (var i = 0; i < 20; i++) {
+          final item = RTreeDatum(Rectangle(0, i, 1, 1), 'Item $i');
+          items.add(item);
+          tree.insert(item);
+        }
+
+        for (final item in items) {
+          tree.remove(item);
+        }
+
+        tree.load(items.sublist(0, 3));
+      });
     });
   });
 }
 
+void assertTreeHeightValidity<E>(RTree<E> tree) {
+  try {
+    assertNodeHeightValidity(tree, tree.currentRootNode);
+  } on StateError catch (e) {
+    fail('${e.message}\nTree:\n${stringifyTree(tree)}');
+  }
+}
+
+int assertNodeHeightValidity<E>(RTree<E> tree, RTreeContributor contributor) {
+  if (contributor is LeafNode<E>) {
+    if (contributor.height != 1) {
+      throw StateError('Leaf height of ${contributor.height} should be 1.');
+    }
+
+    return 1;
+  } else if (contributor is NonLeafNode<E>) {
+    var maxChildHeight = 0;
+    if (contributor.children.isNotEmpty) {
+      for (final child in contributor.children) {
+        final childHeight = assertNodeHeightValidity(tree, child);
+        if (childHeight > maxChildHeight) {
+          maxChildHeight = childHeight;
+        }
+      }
+    }
+
+    final actualNodeHeight = 1 + maxChildHeight;
+    if (contributor.height != actualNodeHeight) {
+      throw StateError('Non-leaf height of ${contributor.height} should be $actualNodeHeight.');
+    }
+
+    return actualNodeHeight;
+  }
+
+  return 0;
+}
+
+String stringifyTree<E>(RTree<E> tree) {
```

Comment:
debug tool left behind, on purpose? probably could use a docstring

Reply:
[This is used by `assertTreeHeightValidity` to print the tree state in the case of a failure.](https://github.com/Workiva/r_tree/blob/2787563c59107d3e3b304e2e44175c71dd468a4f/test/r_tree/r_tree_test.dart#L264) I can add a docstring.